### ISSUE TITLE
Exporting Weights from Tensorflow Keras to C produces a different set of outputs, reduced accuracy in the model.

### ISSUE THREAD

**Issue by BeverlyCode** [keras-team/tf-keras#333]
For some reason when I export my weights from a Tensorflow Keras model, be it a simple Sequential FNN, and then load them into C to perform forward passes I get different output results, sometimes similar and often just not the same at all. It's like the accuracy has taken a substantial loss over the original predict() in Keras.

This is how I export my weights:
```
    for layer in model.layers:
        if layer.get_weights() != []:
            numpy.savetxt(layer.name + ".csv", layer.get_weights()[0].flatten(), delimiter=",")
            numpy.savetxt(layer.name + "_bias.csv", layer.get_weights()[1].flatten(), delimiter=",")
```
and
```
f.write(", /* weight */ " + str(layer.get_weights()[0].flatten()[i]))
f.write(", /* bias */ " + str(layer.get_weights()[1].flatten()[i]))
```
The first method writes a long E value and the second writes a smaller float value to the most significant decimal place. Although that should not cause a loss in precision over the first method? right?

I'm not sure why the saved weights would give different outputs, even with a 1 layer FNN that has 1 output,  a very simple network, all the weights are saved to float32 in C and then processed with float math functions such as tanhf() etc depending on the layer output.

I must be missing some understanding of what Tensorflow or Keras is doing behind the scenes? Something it's default configuration is doing to my network/weights that I need to disable or reproduce? Are the arrays of weights in reverse or something? Do I need to iterate them from left to right and not right to left on the inputs? What am I missing?

Here is a project that demonstrates this issue:
https://github.com/jcwml/neural_unitvector

In this project a dataset is exported from main.c to ensure the C program and Tensorflow Keras program both use data generated by the same random function, then Keras trains a set of weights for a simple FNN network to some number of hidden units and layers, and then those weights are exported and loaded into the C program where the forward pass is reconstructed in C code. There are 3 types of networks in this example that have had their forward passes reconstructed, each more complicated than the last, and yet, none of them produce the same outputs as Keras predict() had, it's not even a minor accuracy loss, it is substantially different, you can tell that the weights still represent what they where trained to but with just much worse prediction accuracy.

Another important fact to note is that when I train a network with multiple layers it would seem in many cases when I export the weights to C for a forward pass that the input data passed into the network eventually becomes 0's before it gets to the deeper layers, like vanishing gradients, this would imply that Tensorflow is using a different precision float format but the documentation says it uses float32 by default and the console printouts also claim to be using float32 in my application. My only thought is that maybe numpy is not saving them in the full float32 precision but when I looked into that apparently it does export full precision, and that exporting weights this way should be fine.

I am very confused.
